docs/remote-access/wireguard.md

```diff
@@ -11,4 +11,112 @@ This page will cover two WireGuard architectures. The first is running WireGuard
 
 ## WireGuard Server on a VPS
 
-* Todo
\ No newline at end of file
+This section will talk about setting up a VPS and then running a client on your network to setup the connection to the VPS and allow the vpn to work without a firewall opening. 
+
+The first step is to decide on a virtual host provider and create a server. When you decide on a provider and create a host. Install Wireguard using the instructions provided on the [Wireguards Documentation](https://www.wireguard.com/install/)
+
+### Server Configuration Setup:
+First create a folder to contain all the details of our configuration.
+```shell
+mkdir wireguard
+cd wireguard
+```
+
+Now you must create the keys you have to options on how many keys you need to make. The simplest allows for just two keys but wireguard also allows you to provide a preshared key as well. For this configuration we will work with the preshared key but if you do not want to you can remove those files from the configuration.
+
+```shell
+wg genkey | tee cloud_key.priv | wg pubkey > cloud_key.pub
+wg genkey | tee client_1_key.priv | wg pubkey > client_1_key.pub
+wg genpsk > client_1_cloud_preshared_key.psk
+```
+
+```shell
+vim /etc/wireguard/wg0.conf
+```
+
+When you create the file you will copy the following configuration:
+```
+[Interface]
+# Cloud
+Address = 192.168.1.1/24
+ListenPort = 51902
+PrivateKey = << Private Key - cloud_key.priv >>
+
+PostUp = iptables -A FORWARD -i %i -j ACCEPT ; iptables -A FORWARD -o %i -j ACCEPT; iptables -t nat -A POSTROUTING -o eth0 -j MASQUERADE
+PostDown = iptables -D FORWARD -i %i -j ACCEPT ; iptables -D FORWARD -o %i -j ACCEPT; iptables -t nat -D POSTROUTING -o eth0 -j MASQUERADE
+
+[Peer]
+# Client 1
+PublicKey = << Public Key - client_1_key.pub >>
+PresharedKey = << Preshared Key - client_1_cloud_preshared_key.psk >>
+AllowedIPs = 192.168.1.2/32,10.0.0.0/16
```

Comment:
The second allowed here is probably unnecessary.

Reply:
I can never quite fully guarantee my understanding of this field. In the server config its purpose is allow the client that IP on the WG net right? You configure routes to specific upstream IPs client side. Right?

Reply:
The configuration this was based off of is a site to site configuration. As mentioned above I like the idea of breaking this into its own section. Maybe just saying add this in the event you are doing a site to site vpn. 

I wanted to include it to show how to add multiple ranges. 

> The last required item for this configuration is the AllowedIPs, to start we will have the IP for the Interface that we will define on the remote client. This is a comma separated list of IPs/subnets.

I included this blurb to try to explain how to fill out the field.  This information maybe better in a table instead of a paragraph so that it is easier to digest. I will include that in my second draft.